src/flows/context/query.tsx

```diff
@@ -112,83 +110,133 @@ export const QueryProvider: FC = ({children}) => {
+    ).map(node => node?.arguments[0]?.properties[0]?.value.value)
+    const orgID =
+      buckets.find(buck => queryBuckets.includes(buck.name))?.orgID || org.id
+
+    // This is all for trying to tease out a window period
+    if (usedVars.hasOwnProperty('windowPeriod')) {
+      const queryRanges = _walk(
+        ast,
+        node =>
+          node?.callee?.type === 'Identifier' && node?.callee?.name === 'range'
+      ).map(node =>
+        (node.arguments[0]?.properties || []).reduce(
+          (acc, curr) => {
+            if (curr.key.name === 'start') {
+              acc.start = propertyTime(ast, curr.value, Date.now())
+            }
+
+            if (curr.key.name === 'stop') {
+              acc.stop = propertyTime(ast, curr.value, Date.now())
+            }
+
+            return acc
+          },
+          {
+            start: '',
+            stop: Date.now(),
           }
-        }
-
-        if (
-          PIPE_DEFINITIONS[pipe.type] &&
-          PIPE_DEFINITIONS[pipe.type].generateFlux
-        ) {
-          PIPE_DEFINITIONS[pipe.type].generateFlux(
-            pipe,
-            create,
-            append,
-            withSideEffects
-          )
+        )
+      )
+
+      if (!queryRanges.length) {
+        ;(((optionAST.body[0] as OptionStatement) // eslint-disable-line no-extra-semi
+          .assignment as VariableAssignment)
+          .init as ObjectExpression).properties.push({
+          type: 'Property',
+          key: {
+            type: 'Identifier',
+            name: 'windowPeriod',
+          },
+          value: {
+            type: 'DurationLiteral',
+            values: [{magnitude: FALLBACK_WINDOW_PERIOD, unit: 'ms'}],
+          },
+        })
+      } else {
+        const starts = queryRanges.map(t => t.start)
+        const stops = queryRanges.map(t => t.stop)
+        const cartesianProduct = starts.map(start =>
+          stops.map(stop => [start, stop])
+        )
+
+        const durations = []
+          .concat(...cartesianProduct)
+          .map(([start, stop]) => stop - start)
+          .filter(d => d > 0)
+
+        const queryDuration = Math.min(...durations)
+        const foundDuration = SELECTABLE_TIME_RANGES.find(
+          tr => tr.seconds * 1000 === queryDuration
+        )
+
+        if (foundDuration) {
+          ;(((optionAST.body[0] as OptionStatement) // eslint-disable-line no-extra-semi
+            .assignment as VariableAssignment)
+            .init as ObjectExpression).properties.push({
+            type: 'Property',
+            key: {
+              type: 'Identifier',
+              name: 'windowPeriod',
+            },
+            value: {
+              type: 'DurationLiteral',
+              values: [{magnitude: foundDuration.windowPeriod, unit: 'ms'}],
+            },
+          })
         } else {
-          append()
+          ;(((optionAST.body[0] as OptionStatement) // eslint-disable-line no-extra-semi
```

Comment:
what's the point of the `;` here?

Reply:
prettier adds it to.. escape execution? idk, the robots were fighting over it. one added it, one hated it and tried to remove it, i just told them to settle it down

Reply:
huh how bizarre. Well if the robots say it's correct, then I'm okay with it!

![](https://media.giphy.com/media/obWh8XqeiaT8A/giphy.gif)